src/source/RazorEngine.Core/Templating/TemplateBase.cs

```diff
@@ -237,23 +240,48 @@ void ITemplate.Run(ExecuteContext context, TextWriter reader)
 
                     if (layout == null)
                     {
-                        throw new ArgumentException("Template you are trying to run uses layout, but no layout found in cache or by resolver.");
+                        throw new ArgumentException(
+                            "Template you are trying to run uses layout, but no layout found in cache or by resolver.");
                     }
 
-                    // Push the current body instance onto the stack for later execution.
-                    var body = new TemplateWriter(tw => tw.Write(builder.ToString()));
-                    context.PushBody(body);
+                    var body = new TemplateWriter(
+                        tw =>
+                            {
+                                using (var reader = new StreamReader(tempFilePath))
+                                {
+                                    // Push the current body instance onto the stack for later execution.
+                                    while ( !reader.EndOfStream)
+                                    {
+                                        tw.Write(reader.ReadLine());
```

Comment:
`ReadLine` does not include the new line characters. That's why some tests are failing.